hazelcast-jet-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/JoinScanProcessorSupplier.java

```diff
@@ -0,0 +1,140 @@
+import com.hazelcast.jet.sql.impl.connector.keyvalue.KvRowProjector;
+import com.hazelcast.map.IMap;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.query.impl.getters.Extractors;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map.Entry;
+
+@SuppressFBWarnings(
+        value = {"SE_BAD_FIELD", "SE_NO_SERIALVERSIONID"},
+        justification = "the class is never java-serialized"
+)
+final class JoinScanProcessorSupplier implements ProcessorSupplier, DataSerializable {
+
+    private static final int MAX_BATCH_SIZE = 1024;
+
+    private JetJoinInfo joinInfo;
+    private String mapName;
+    private KvRowProjector.Supplier rightRowProjectorSupplier;
+
+    private transient IMap<Object, Object> map;
+    private transient InternalSerializationService serializationService;
+    private transient Extractors extractors;
+
+    @SuppressWarnings("unused")
+    private JoinScanProcessorSupplier() {
+    }
+
+    JoinScanProcessorSupplier(
+            JetJoinInfo joinInfo,
+            String mapName,
+            KvRowProjector.Supplier rightRowProjectorSupplier
+    ) {
+        this.joinInfo = joinInfo;
+        this.mapName = mapName;
+        this.rightRowProjectorSupplier = rightRowProjectorSupplier;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) {
+        map = context.jetInstance().getMap(mapName);
+        serializationService = ((ProcSupplierCtx) context).serializationService();
+        extractors = Extractors.newBuilder(serializationService).build();
+    }
+
+    @Nonnull
+    @Override
+    public Collection<? extends Processor> get(int count) {
+        List<Processor> processors = new ArrayList<>(count);
+        for (int i = 0; i < count; i++) {
+            Processor processor = new TransformBatchedP<Object[], Object[]>(
+                    MAX_BATCH_SIZE,
+                    joinFn(joinInfo, map, rightRowProjectorSupplier.get(serializationService, extractors))
+            ) {
+                @Override
+                public boolean isCooperative() {
+                    return false;
+                }
+            };
+            processors.add(processor);
+        }
+        return processors;
+    }
+
+    private static FunctionEx<List<? super Object[]>, Traverser<Object[]>> joinFn(
+            JetJoinInfo joinInfo,
+            IMap<Object, Object> map,
+            KvRowProjector rightRowProjector
+    ) {
+        BiFunctionEx<Object[], Object[], Object[]> joinFn = ExpressionUtil.joinFn(joinInfo.condition());
+
+        return lefts -> {
+            List<Object[]> rows = new ArrayList<>();
+            for (Entry<Object, Object> entry : map.entrySet()) {
```

Comment:
We should:
- cache the result `entrySet` for some time (like 10 seconds). We should cache it projected.
- share it among local processors: maybe have a volatile field in the PSupplier and have 0th local processor update it for everyone.